shared/desktop/remote/sync-avatar-props.desktop.tsx

```diff
@@ -56,21 +54,28 @@ export const deserialize = (state: any = initialState, props: any) => {
 }
 
 function SyncAvatarProps(ComposedComponent: any) {
-  class RemoteAvatarConnected extends React.PureComponent<Props> {
-    render() {
-      const {setUsernames, usernames, ...rest} = this.props
-      return <ComposedComponent {...rest} />
-    }
+  const RemoteAvatarConnected = (props: Props) => {
+    const {usernamesRef, ...rest} = props
+    return <ComposedComponent {...rest} />
   }
 
-  const mapStateToProps = (state: Container.TypedState, ownProps: OwnProps) => ({
-    ...immutableCached(
-      getRemoteFollowers(state.config.followers, ownProps.usernames),
-      getRemoteFollowing(state.config.following, ownProps.usernames)
-    ),
-    httpSrvAddress: state.config.httpSrvAddress,
-    httpSrvToken: state.config.httpSrvToken,
-  })
+  const mapStateToProps = (state: Container.TypedState, ownProps: OwnProps) => {
+    const {
```

Comment:
i find this nested destructuring hard to read. i think its clearer if you do it one level at a time aka
```
const {usernamesRef} = ownProps
const {current: oldUsername} = usernamesRef
```

Reply:
This took me a long time to parse, I agree.

Reply:
Sure thing 👍